src/components/PostCard/components/MediaPlayer/hooks/useSlider.ts

```diff
@@ -0,0 +1,62 @@
+import React, { useEffect, useRef, useState } from "react"
+
+const useSlider = ({ initialValue }: { initialValue: number }) => {
+  const sliderRef = useRef<HTMLDivElement>(null)
+  const [isDragging, setIsDragging] = useState(false)
+  const [newValue, setNewValue] = useState(initialValue)
+
+  const handleMouseDown = ({ clientX }: React.MouseEvent) => {
+    setIsDragging(true)
+
+    const { current } = sliderRef
+    if (!current) {
+      return
+    }
+
+    const rect = current.getBoundingClientRect()
+    const handlerOffset = clientX - rect.left
+
+    const calcValue = handlerOffset / current.offsetWidth
+    let checkedValue = Math.max(0, calcValue)
+    checkedValue = Math.min(1, checkedValue)
+
+    setNewValue(checkedValue)
+  }
+
+  const handleMouseUp = () => {
+    setIsDragging(false)
+  }
+
+  useEffect(() => {
+    const handleMouseMove = ({ clientX }: MouseEvent) => {
+      if (!isDragging) {
+        return
+      }
+      const { current } = sliderRef
+      if (!current) {
+        return
+      }
+
+      const rect = current.getBoundingClientRect()
+      const handlerOffset = clientX - rect.left
+
+      const calcValue = handlerOffset / current.offsetWidth
+      let checkedValue = Math.max(0, calcValue)
+      checkedValue = Math.min(1, checkedValue)
```

Comment:
여기는 이런식으로 줄여볼 수 있을 것 같습니다👍
```suggestion
    const checkedValue = Math.min(1, Math.max(0, calcValue))
```

Reply:
해당 부분은 가독성측면에 좋지않아서 분리하였습니다